lib/url.js

```diff
@@ -10,60 +10,73 @@ exports.format = urlFormat;
 exports.Url = Url;
 
 function Url() {
-  this.protocol = null;
-  this.slashes = null;
+  // For more efficient internal representation and laziness.
+  // The non-underscore versions of these properties are accessor functions
+  // defined on the prototype.
+  this._protocol = null;
+  this._href = '';
+  this._port = -1;
+  this._query = null;
+  this._prependSlash = false;
   this.auth = null;
+  this.slashes = null;
   this.host = null;
-  this.port = null;
   this.hostname = null;
```

Comment:
I'm concerned about the inconsistency of exposing half the properties as getters and half as instance properties. I think it should be all one or the other.